notes/stv/20221130-test-deploy-01.txt

```diff
@@ -0,0 +1,566 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2022, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
+    Target:
+
+        Test deploy, with changes to Nginx Proxy for:
+           - Fixing issue with Safari / login 
+           - Setting up an HTTPS dev system on red via the deployment scripts
+
+    Result:
+  
+        Success
+
+# -----------------------------------------------------
+# Check which cloud is currently live.
+#[user@desktop]
+
+    ssh fedora@live.gaia-dmp.uk \
+        '
+        date
+        hostname
+        '
+
+    >   iris-gaia-blue-20221013-zeppelin
+
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    # Use Red for deploy
+    
+    source "${HOME:?}/aglais.env"
+
+    agcolour=red
+    configname=zeppelin-26.43-spark-3.26.43
+
+    agproxymap=3000:3000
+    clientname=ansibler-${agcolour}
+    cloudname=iris-gaia-${agcolour}
+
+    podman run \
+        --rm \
+        --tty \
+        --interactive \
+        --name     "${clientname:?}" \
+        --hostname "${clientname:?}" \
+        --publish  "${agproxymap:?}" \
+        --env "cloudname=${cloudname:?}" \
+        --env "configname=${configname:?}" \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK:?}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        ghcr.io/wfau/atolmis/ansible-client:2022.07.25 \
+        bash
+
+
+# -----------------------------------------------------
+# Deploy everything.
+#[root@ansibler]
+
+    time \
+        source /deployments/hadoop-yarn/bin/deploy.sh iris-gaia-red.gaia-dmp.uk
```

Comment:
Why do we have to manually specify the hostname here? The deployment script should be able to work this out based on `${cloudname}`.
```suggestion
        source /deployments/hadoop-yarn/bin/deploy.sh
```

Reply:
Also are we then essentially hard coding the domain name in the deployment scripts, i.e.  iris-gaia-{color}.gaia-dmp.uk ? If we want to use a different hostname, we'd have to remember to change that

Reply:
Why do any of the deployments need to know their host name ?

Reply:
All of the deployments should be able to handle all of the hostnames.
See #838

Reply:
> I can change this so that it picks it up from the cloudname, but if so in the case of a public deploy, where do we specify that the proxy should be setup with dmp.gaia.ac.uk as the hostname?

All of the deployments should be configured to handle all of the hostnames, including the public address(es). We should not need to have different steps for the live deployment.

Reply:
See #1066

Reply:
See #1067

Reply:
Might be worth splitting this PR into one that fixes the user facing issue #1057, and then address the dns names in a separate PR later.

Reply:
Ok I've removed the changes specific to setting up the hostname, and just left the fixes to the proxy on this branch